toolchain/check/testdata/class/fail_abstract.carbon

```diff
@@ -20,16 +20,22 @@ class Derived {
 
 fn Make() -> Derived {
   // TODO: This should be valid, and should construct an instance of `partial Abstract` as the base.
-  // CHECK:STDERR: fail_abstract.carbon:[[@LINE+3]]:19: error: cannot construct instance of abstract class; consider using `partial Abstract` instead
+  // CHECK:STDERR: fail_abstract.carbon:[[@LINE+4]]:19: error: cannot construct instance of abstract class; consider using `partial Abstract` instead
   // CHECK:STDERR:   return {.base = {.a = 1}, .d = 7};
   // CHECK:STDERR:                   ^~~~~~~~
+  // CHECK:STDERR:
   return {.base = {.a = 1}, .d = 7};
 }
 
 fn Access(d: Derived) -> (i32, i32) {
   return (d.d, d.base.a);
 }
 
+// CHECK:STDERR: fail_abstract.carbon:[[@LINE+3]]:1: error: `abstract` not allowed on `class` declaration
```

Comment:
It would be nice (but not required) to have the error say this modifier should only be on the definition.

Reply:
Yeah, it crossed my mind when developing this, but the generic functionality for disallowed modifiers was rather convenient & adding this special case felt like it'd be pretty ugly.

I've implemented it/updated this patch with that change just for the exploration - maybe there's nicer ways to do it (with a callback?) or perhaps it feels worth the complexity/inelegance. Open to ideas/refactorings/revertings/renamings/etc.